src/actions/LoginInitActions.tsx

```diff
@@ -325,6 +325,8 @@ export const checkAndRequestNotifications = (branding: Branding) => async (
       bridge={bridge}
       message={permissionMessage}
       choices={choiceMap}
+      notifEnabled={notifEnabled === 1}
+      onLogEvent={onLogEvent}
```

Comment:
I'd argue that these data types are relevant to the component. If we had the pattern of logging events at the callsites of modals, then we would need to remember at each new callsite that we also need to log the event, which could potentially be missed by reviews.

Since this is only called once though I'll make the change as suggested

Reply:
> notifEnabled is a condition because the modal isn't necessarily always requesting notification permissions. We are only interested in the event that they enabled notifications via this modal.

Thanks for explaining the reason why `notifEnabled === 0` is used, now I understand the logic. I'd add the comment to this condition "only log if the notifications were not enabled beforehand" or something like that.

> If we had the pattern of logging events at the callsites of modals, then we would need to remember at each new callsite that we also need to log the event, which could potentially be missed by reviews.

This same argument can be applied to all the other logic at the call-site though. We have one case of the modal component being implemented today, and should we implement it elsewhere all the logic at the call-site should be considered for that implementation. Where and why would we even have two places for this modal? What if at that place we don't want to log the event? Why don't we consider what this component looks like when we actually have another use case for this modal component, and consider what it's responsibilities are at that time?

Reply:
Had a sidebar on requirements and actually it's better here. We're going to be interested in both 
1. Whether they have notifications enabled for any conversion event, doesn't matter how notifications got enabled
2. The efficacy of this modal specifically increasing the rate of allowing notification permissions